tests/test_state_client.cpp

```diff
@@ -42,10 +43,23 @@ TEST_F(smTests, testSMSetStateOff)
 
 TEST_F(smTests, testSMSetStateUndefined)
 {
+    ara::exec::MockStateClient MyStateClient;
+    mySM.stateClient = &MyStateClient;
+    EXPECT_CALL(MyStateClient, undefinedStateCallback);
+
     EXPECT_EQ(mySM.internalState, ara::sm::FunctionGroupStateType::Off);
-    mySM.stateClient->SmSetState((ara::sm::FunctionGroupStateType) ara::sm::MachineStateType::Startup);
+    mySM.stateClient->SmSetState((ara::sm::FunctionGroupStateType) ara::sm::MachineStateType::Restart);
     // let changes to be applied
     using namespace std::chrono_literals;
     std::this_thread::sleep_for(500ms);
     EXPECT_EQ(mySM.internalState, ara::sm::FunctionGroupStateType::Off);
 }
+
+TEST_F(smTests, testInitialStateTransitionResult)
+{
+    mySM.stateClient->SmSetState(ara::sm::FunctionGroupStateType::On);
+    // let changes to be applied
+    using namespace std::chrono_literals;
+    std::this_thread::sleep_for(500ms);
+    EXPECT_EQ(mySM.stateClient->GetInitialMachineStateTransitionResult(), ara::sm::ErrorType::kSuccess);
```

Comment:
I would suggest only using EXPECT_CALL here, this method shall be called from SM

Reply:
Test body changed. Is this comment still appropriate?